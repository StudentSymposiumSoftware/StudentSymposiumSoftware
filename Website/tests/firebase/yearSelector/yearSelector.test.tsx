import { expect, it } from "vitest";
import React from 'react';
import { render } from "@testing-library/react";

import YearSelector from '../../../src/firebase/yearSelector';

it('yearSelector renders correctly', () => {
    const availableYears = ["2025", "2024", "2023", "2022", "2021"];
    const tree = render(<YearSelector setYear={null} availableYears={availableYears} currYear={"2025"}/>)
    expect(tree).toMatchSnapshot();
});
