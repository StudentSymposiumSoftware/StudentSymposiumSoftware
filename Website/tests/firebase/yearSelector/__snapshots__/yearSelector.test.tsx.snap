// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`yearSelector renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white text-black w-[100%]"
      >
        <div>
          Select Year of Symposium Data to Review:
        </div>
        <div>
          <select
            class="bg-white text-black w-min cursor-pointer"
          >
            <option
              selected=""
              value="2025"
            >
              2025
            </option>
            <option
              value="2024"
            >
              2024
            </option>
            <option
              value="2023"
            >
              2023
            </option>
            <option
              value="2022"
            >
              2022
            </option>
            <option
              value="2021"
            >
              2021
            </option>
          </select>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white text-black w-[100%]"
    >
      <div>
        Select Year of Symposium Data to Review:
      </div>
      <div>
        <select
          class="bg-white text-black w-min cursor-pointer"
        >
          <option
            selected=""
            value="2025"
          >
            2025
          </option>
          <option
            value="2024"
          >
            2024
          </option>
          <option
            value="2023"
          >
            2023
          </option>
          <option
            value="2022"
          >
            2022
          </option>
          <option
            value="2021"
          >
            2021
          </option>
        </select>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
